File: typescript/api/src/responseBuilder.ts
import {GameStateResponse} from "./response/gameStateResponse";
import {Room} from "./base/room";
import {GameEntityResponse} from "./response/gameEntityResponse";

/**
 * Builder class to  build responseObjects (that are send to client)
 *
 */
export class ResponseBuilder {

    /**
     * Handle a custom action
     *
     * @param room where the gamers is located
     * @param gameStateResponse GameStateResponse that need to be filled
     *
     */
    public static fillResponse(room: Room, gameState: GameStateResponse): void {
        const roomObject: GameEntityResponse = new GameEntityResponse();
        roomObject.fullDescription = room.getFullDescription();
        roomObject.shortDescription = room.getShortDescription();
        roomObject.id = room.id;
        gameState.currentRoom = roomObject;
        room.objects.forEach(function (object) {
            const objectInRoom: GameEntityResponse = new GameEntityResponse();
            objectInRoom.id = object.id;
            objectInRoom.fullDescription = object.getFullDescription();
            objectInRoom.shortDescription = object.getShortDescription();
            gameState.objectsInRoom.push(objectInRoom);
        });
        return;
    }


}

File: java/api/src/main/java/game/engine/java/api/ResponseBuilder.java
package game.engine.java.api;

import game.engine.java.api.base.GameObject;
import game.engine.java.api.base.Room;
import game.engine.java.api.response.GameEntityResponse;
import game.engine.java.api.response.GameStateResponse;

public class ResponseBuilder {

    public static void fillResponse(Room room, GameStateResponse gameState) {
        GameEntityResponse roomObject = new GameEntityResponse();
        roomObject.setFullDescription(room.getFullDescription());
        roomObject.setShortDescription(room.getShortDescription());
        roomObject.setId(room.getId());
        gameState.setCurrentRoom(roomObject);
        for (GameObject object : room.getObjects()) {
            GameEntityResponse gameEntityResponse = new GameEntityResponse();
            gameEntityResponse.setId(object.getId());
            gameEntityResponse.setFullDescription(object.getFullDescription());
            gameEntityResponse.setShortDescription(object.getShortDescription());
            gameState.getObjectsInRoom().add(gameEntityResponse);
        }
    }
}
